CONTRIBUTING.md

```diff
@@ -0,0 +1,103 @@
+# So You Want to Contribute to Papermill...
+We welcome all contributions to Papermill both large and small. We encourage you
+to join our community.
+
+## Our Community Values
+
+We are an open and friendly community. Everybody is welcome.
+
+We encourage friendly discussions and respect for all. There are no exceptions.
+
+All contributions are equally important. Documentation, answering questions, and
+fixing bugs are equally as valuable as adding new features.
+
+Check out <https://www.python.org/psf/codeofconduct/> for the Python code of conduct.
+
+## Setting up Your Development Environment
+Following these instructions should give you an efficient path to opening your first pull-request.
+
+### Cloning the Papermill Repository
+Fork the repository to your local Github account. Clone this repository to your local development machine.
+```
+git clone https://github.com/<your_account>/papermill
+cd papermill
+```
+
+### Install an Editable Version
+We prefer to use [conda](https://conda.io/docs/user-guide/tasks/manage-environments.html) to manage the development environment. 
+```
+conda create -n dev
+. activate env
+``` 
+or [virtualenv](https://packaging.python.org/guides/installing-using-pip-and-virtualenv/) if you prefer.
+```
+python3 -m virtualenv dev
+source dev/bin/activate 
+```
+
+```
+pip install -e .
+```
+
+_Note: When you are finished you can use `source deactivate` to go back to your base environment._
+
+### Running Tests Locally
+We need to install the development package before we can run the tests. If anything is confusing below,
+always resort to the relevant documentation.
+```buildoutcfg
+pip install papermill[dev]
+py.test --pyargs papermill
+```
+
+Now there should be a working and editable installation of Papermill to start making your own contributions.
+## So You're Ready to Pull Request
+The general workflow for this will be:
+1. Run local tests 
+2. Pushed changes to your forked repository
+3. Open pull request to main repository
+
+### Run Tests Locally
+```
+py.test --pyargs papermill
```

Comment:
While you can also run pytest from anywhere within the directory, this actually tests the module and is following pytest's [best practices](https://docs.pytest.org/en/latest/goodpractices.html#tests-as-part-of-application-code) which I hadn't realised was a best practice (just a useful way of being able to run this anywhere). 

I would include that link with a brief explanation of what `--pyargs` is doing when you run the tests.